src/ErrorHandler.php

```diff
@@ -54,11 +63,33 @@ public function addFormatters(FormatterInterface ...$formatters): self
         return $this;
     }
 
+    /**
+     * Set a custom log callback
+     */
+    public function logCallback(callable $callback): self
```

Comment:
I think `logger()` could accept a callable or a LoggerInterface, so we don't need two methods for the same purpose.
Other option is `->logger($logger, $callback)`, but to me is more elegant to have only one handler for logging (the callback can have access to the logger by itself).

Reply:
ok, but then it wouldn't be about logging anymore, but a hook for when an error happens, right? Because inside the callback you can put anything, there is no LoggerInterface enforcement anymore so we should change the name.

How does this look like? `onError`

```php
$errorHandler->onError(function(Throwable $error, ServerRequestInterface $request) use($logger, $other): void {
    $logger->error('Whoops!');

    // $other stuff...
});
```

Reply:
It looks good to me. Maybe an `onError` callback is enough to cover all casuistry and we don't need a specific method for logger.